### ISSUE TITLE
ExodusII IO has variable ID bug which can prevent coordinates from being read-in from a file

### ISSUE THREAD

**Issue by EdwardPalmer99** [libMesh/libmesh#3823]
**Brief**
- Version 5.22b of the exodusii library contains a bug in the ex_get_coord function. Note that this issue is not present in version v8.11.

**What is the problem?**
- NetCDF variable IDs are numbered starting from 0 in C (from 1 in Fortran) [https://docs.unidata.ucar.edu/netcdf-c/current/tutorial_ncids.html](url)
- Version 5.22b contains a bug in the file ex_get_coord.c on line 238 where if the variable ID of a coordinate is 0 (which is a valid variable ID!) then the coordinate data will not be read from the file. This will result in one of the coordinates being an array of zeros. This was a problem that I encountered when reading in certain ExodusII files.
- This bug has been addressed in more recent versions of the ExodusII library by comparing against a value of (-1) which is not a valid variable ID. I recommend updating the exodusii reader or changing this line in the source code.

**Links**
- Link to problematic file (v5.22b): [https://github.com/libMesh/libmesh/blob/7dbd90ade88cb16a5e855a4c345454c9c146ebec/contrib/exodusii/5.22b/exodus/cbind/src/ex_get_coord.c](url)

- Link to updated file in seacas repository with fix: [https://github.com/sandialabs/seacas/blob/master/packages/seacas/libraries/exodus/src/ex_get_coord.c](url)

**Code**
`
        if (coor != NULL && coordid != 0) 
`

**Comment by jwpeterson** [comment]
Hi @EdwardPalmer99, 

Thanks for the info. For quite a while now, I believe we default to using Exodus v8.11 in libMesh, so hopefully this bug does not affect many of our users. I'm curious, if you are actually using libmesh and still getting Exodus 5.22b, what version of libmesh are you using and how is it configured?

We could try and patch the 5.22 sources that we bundle with libmesh, but at this point it might be more productive to actually remove them from master and make sure that everyone is using v8.11.

**Comment by EdwardPalmer99** [comment]
Hi, the libmesh commit hash is _7dbd90ade88cb16a5e855a4c345454c9c146ebec_ (2nd April 2024). This is the version currently in use by the master branch of Moose. Libmesh was built without the optional HDF5 library which may be why the older 5.22 Exodus version was used.
![Screenshot from 2024-04-05 11-56-07](https://github.com/libMesh/libmesh/assets/93386624/8a9ada53-b3d9-456b-835a-9e8acc6d4d80)
![Screenshot from 2024-04-05 11-52-41](https://github.com/libMesh/libmesh/assets/93386624/c218a0bb-9856-4410-b8d4-837f69fdffb4)

**Comment by EdwardPalmer99** [comment]
Yes, this is a simple ExodusII file. It can be read-in successfully using Coreform Cubit. Please note that I have changed the extension to 'txt' in order to upload it to Github.

[simple-cube-hex8-dodgy.txt](https://github.com/libMesh/libmesh/files/14932796/simple-cube-hex8-dodgy.txt)
